pkg/controllers/ttl/controller.go

```diff
@@ -43,19 +47,21 @@ func newController(client metadata.Getter, metainformer informers.GenericInforme
 		client:   client,
 		queue:    workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
 		lister:   metainformer.Lister(),
-		wg:       wait.Group{},
 		informer: metainformer.Informer(),
 		logger:   logger,
 		metrics:  newTTLMetrics(logger),
 	}
-	registration, err := c.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    c.handleAdd,
-		UpdateFunc: c.handleUpdate,
-	})
+	registration, err := controllerutils.AddEventHandlersT(
+		c.informer,
+		c.handleAdd,
+		c.handleUpdate,
+		c.handleDelete,
```

Comment:
not sure we want to enqueue on delete

Reply:
The controller is panicking after the deletion of the resource in case of non-inclusion of the handleDelete() function.

Reply:
this doesn't mean `handleDelete` has to enqueue the item.

Reply:
but I think this should be good as the user will get to know that which resource is being deleted in the logs if he wants to.

Reply:
I fixed the controller utils package

Reply:
Yes you're correct, fixing it now.

Reply:
I fixed it at the root.